statsd/buffer_pool.go

```diff
@@ -0,0 +1,30 @@
+package statsd
+
+import (
+	"bytes"
+	"sync"
+)
+
+type bufferPool struct {
+	sync.Pool
+}
+
+func newBufferPool() *bufferPool {
+	bp := &bufferPool{}
+
+	bp.New = func() interface{} {
+		return &bytes.Buffer{}
+	}
+
+	return bp
+}
+
+func (bp *bufferPool) Get() *bytes.Buffer {
+	b := (bp.Pool.Get()).(*bytes.Buffer)
+	b.Truncate(0)
```

Comment:
I prefer to have the truncate/reset happen in the Put (right before bp.Pool.Put of course) instead of in the Get. That way you can avoid calling truncate for the cases where the sync.Pool ends up actually calling New.